### PR TITLE
[VirtualWan] Support generate and download P2S VPN client

### PR DESCRIPTION
**Pull request by haroldrandom**
Close https://github.com/Azure/azure-cli-extensions/issues/2465

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
VirtualWan

File: src/virtual-wan/azext_vwan/commands.py [hunk 1]
```diff
@@ -196,4 +196,7 @@ def load_command_table(self, _):
     with self.command_group('network p2s-vpn-gateway connection', network_util, min_api='2020-04-01', is_preview=True) as g:
         g.command('list', list_network_resource_property(resource, prop))
         g.show_command('show', get_network_resource_property_entry(resource, prop))
+
+    with self.command_group('network p2s-vpn-gateway vpn-client', network_p2s_vpn_gateway_sdk) as g:
```
Comment (Juliehzl) [inline src/virtual-wan/azext_vwan/commands.py:200]: does the command group also have min api restriction?

**Review by Juliehzl — COMMENTED** [review]
should we also need to update history.rst for release notw update?

File: src/virtual-wan/azext_vwan/_params.py [hunk 2]
```diff
@@ -212,6 +212,9 @@ def load_arguments(self, _):
 
     with self.argument_context('network p2s-vpn-gateway connection list') as c:
         c.argument('resource_name', p2s_vpn_gateway_name_type, id_part=None)
+
+    with self.argument_context('network p2s-vpn-gateway vpn-client') as c:
+        c.argument('authentication_method', arg_type=get_enum_type(AuthenticationMethod))
```
Comment (kairu-ms) [inline src/virtual-wan/azext_vwan/_params.py:217]: can you use '--auth-method' for short?

File: scripts/ci/credscan/CredScanSuppressions.json [hunk 3]
```diff
@@ -86,6 +86,12 @@
         "src\\communication\\azext_communication\\tests\\latest\\test_communication_scenario.py"
       ],
       "_justification": "Dummy resources' tokens left during testing Micorosfot.Communication"
+    },
+    {
+      "file": [
+        "src\\virtual-wan\\azext_vwan\\tests\\latest\\recordings\\test_p2s_vpn_client_generate.yaml"
+      ],
+      "_justification": "download URL to non-existing VPN client"
```
Comment (kairu-ms) [inline scripts/ci/credscan/CredScanSuppressions.json:94]: You should use SAS replacer in test.
FYI
https://github.com/Azure/azure-cli-extensions/blob/549d71be94bc3e31c43fb218a571d5d4bf213ab3/src/application-insights/azext_applicationinsights/tests/latest/recording_processors.py#L19-L46
https://github.com/Azure/azure-cli-extensions/blob/549d71be94bc3e31c43fb218a571d5d4bf213ab3/src/application-insights/azext_applicationinsights/tests/latest/test_applicationinsights_mgmt.py#L13-L17

File: src/virtual-wan/azext_vwan/commands.py (hunk 1 above)
Comment (haroldrandom) [reply src/virtual-wan/azext_vwan/commands.py]: added

**Comment by haroldrandom** [comment]
> should we also need to update history.rst for release notw update?

updated

File: src/virtual-wan/azext_vwan/_params.py (hunk 2 above)
Comment (haroldrandom) [reply src/virtual-wan/azext_vwan/_params.py:217]: I followed the style `az network vnet-gateway vpn-client generate -h`. It used `--authentication-method`.
But I am okay with it. we could change it to `--auth-method`

File: scripts/ci/credscan/CredScanSuppressions.json (hunk 3 above)
Comment (haroldrandom) [reply scripts/ci/credscan/CredScanSuppressions.json]: done

**Review by kairu-ms — APPROVED** [review]

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**